CMakeLists.txt

```diff
@@ -0,0 +1,155 @@
+PROJECT(Snappy C CXX)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.0)
+
+SET(SNAPPY_MAJOR 1)
+SET(SNAPPY_MINOR 1)
+SET(SNAPPY_PATCHLEVEL 3)
+
+# I haven't looked at what the soname for the previous
+# versions of snappy... let's just start with 1.1.3
+SET(SNAPPY_SONAME "1.1.3")
+
+INCLUDE(CheckIncludeFiles)
+INCLUDE(CheckLibraryExists)
+INCLUDE(CheckCXXSourceCompiles)
+INCLUDE(TestBigEndian)
+INCLUDE(GenerateExportHeader)
+INCLUDE(CMakePackageConfigHelpers)
+
+TEST_BIG_ENDIAN(WORDS_BIG_ENDIAN)
+IF (WORDS_BIG_ENDIAN)
+    MESSAGE(STATUS "Builing on big endian system")
+    ADD_DEFINITIONS(-DWORDS_BIGENDIAN=1)
+ENDIF (WORDS_BIG_ENDIAN)
+
+CHECK_INCLUDE_FILES("windows.h" HAVE_WINDOWS_H)
+CHECK_INCLUDE_FILES("byteswap.h" HAVE_BYTESWAP_H)
+CHECK_INCLUDE_FILES("dlfcn.h" HAVE_DLFCN_H)
+CHECK_INCLUDE_FILES("inttypes.h" HAVE_INTTYPES_H)
+CHECK_INCLUDE_FILES("memory.h" HAVE_MEMORY_H)
+CHECK_INCLUDE_FILES("stddef.h" HAVE_STDDEF_H)
+CHECK_INCLUDE_FILES("stdint.h" HAVE_STDINT_H)
+CHECK_INCLUDE_FILES("stdlib.h" HAVE_STDLIB_H)
+CHECK_INCLUDE_FILES("strings.h" HAVE_STRINGS_H)
+CHECK_INCLUDE_FILES("string.h" HAVE_STRING_H)
+CHECK_INCLUDE_FILES("sys/byteswap.h" HAVE_SYS_BYTESWAP_H)
+CHECK_INCLUDE_FILES("sys/endian.h" HAVE_SYS_ENDIAN_H)
+CHECK_INCLUDE_FILES("stdint.h" HAVE_STDINT_H)
+CHECK_INCLUDE_FILES("sys/mman.h" HAVE_SYS_MMAN_H)
+CHECK_INCLUDE_FILES("sys/resource.h" HAVE_SYS_RESOURCE_H)
+CHECK_INCLUDE_FILES("sys/stat.h" HAVE_SYS_STAT_H)
+CHECK_INCLUDE_FILES("sys/time.h" HAVE_SYS_TIME_H)
+CHECK_INCLUDE_FILES("sys/types.h" HAVE_SYS_TYPES_H)
+CHECK_INCLUDE_FILES("unistd.h" HAVE_UNISTD_H)
+CHECK_INCLUDE_FILES("sys/uio.h" HAVE_SYS_UIO_H)
+
+IF (NOT HAVE_SYS_UIO_H)
+    SET(HAVE_SYS_UIO_H 0)
+ENDIF (NOT HAVE_SYS_UIO_H)
+
+IF (NOT HAVE_STDINT_H)
+    SET(HAVE_STDINT_H 0)
+ENDIF (NOT HAVE_STDINT_H)
+
+IF (NOT HAVE_STDDEF_H)
+    SET(HAVE_STDDEF_H 0)
+ENDIF (NOT HAVE_STDDEF_H)
+
+CHECK_LIBRARY_EXISTS(z zlibVersion "" HAVE_LIBZ)
+CHECK_LIBRARY_EXISTS(lzo2 lzo1x_1_15_compress "" HAVE_LIBLZO2)
+CHECK_LIBRARY_EXISTS(lzf lzf_compress "" HAVE_LIBLZF)
+CHECK_LIBRARY_EXISTS(fastlz fastlz_compress "" HAVE_LIBFASTLZ)
+CHECK_LIBRARY_EXISTS(quicklz qlz_compress "" HAVE_LIBQUICKLZ)
+
+CHECK_CXX_SOURCE_COMPILES("int main(void) { return __builtin_expect(0, 1) ? 1 : 0; }"
+        HAVE_BUILTIN_EXPECT)
+
+CHECK_CXX_SOURCE_COMPILES("int main(void) { return __builtin_ctzll(0) ? 1 : 0; }"
+        HAVE_BUILTIN_CTZ)
+
+
+CONFIGURE_FILE(${Snappy_SOURCE_DIR}/cmake/config.h.in config.h)
+CONFIGURE_FILE(${Snappy_SOURCE_DIR}/cmake/snappy-stubs-public.h.in snappy-stubs-public.h)
+
+IF (WIN32)
+    ADD_DEFINITIONS(-D_CRT_SECURE_NO_WARNINGS)
+ENDIF (WIN32)
+
+ADD_DEFINITIONS(-DHAVE_CONFIG_H)
+
+INCLUDE_DIRECTORIES(BEFORE ${Snappy_SOURCE_DIR})
```

Comment:
Forgive my Cmake noobishness, but Cmake [best practices] (http://www.slideshare.net/DanielPfeifer1/cmake-48475415) says do not use include_directories but target_include_directories instead.